### PR TITLE
feat(all): update node dependencies

### PR DESCRIPTION
**Pull request by m1ga**
update some node dependencies to have a newer babel version that includes a newer browserlist/canuse-lite version.

Resolves https://github.com/tidev/titanium_mobile/issues/13286

**Test**
* clone
* npm i
* copy the whole folder into your "sdk-version" `/node_modules/node-titanium-sdk/` folder. 
* build apps

I'm not seeing the `npx browserslist@latest --update-db` message after `[INFO]  Processing JavaScript files
` Apps still build fine (Kitchensink, hyperloop examples, own app)

After that we'll need to update https://github.com/tidev/titanium_mobile/blob/20d2ed1806d1cedd58f226a70b384997c5c5ac9a/package.json#L104

### CODE REVIEW

**Comment by m1ga** [comment]
tests fail because the base64 encoded lines are in a different order :thinking:

**Review by cb1kenobi — APPROVED** [review]

**Comment by cb1kenobi** [comment]
@m1ga I see I approved this PR, but never merged it!? Sigh. Since this PR is a bit old, would you mind refreshing the dependencies quick and hopefully the checks will pass?

**Comment by m1ga** [comment]
Thought I had the test order fixed but it looks like it's different again. Will check that later

File: lib/jsanalyze.js [hunk 1]
```diff
@@ -26,6 +26,23 @@ const path = require('path');
 const SOURCE_MAPPING_URL_REGEXP = /\/\/#[ \t]+sourceMappingURL=([^\s'"`]+?)[ \t]*$/mg;
 const __ = appc.i18n(__dirname).__;
 
+function sortObject (o) {
+	var sorted = {},
+		key,
+		a = [];
+
+	for (key in o) {
```
Comment (cb1kenobi) [inline lib/jsanalyze.js:34]: In general, I think this function can be simplified to something like:

```
const sorted = {};
for (const key of Object.keys(o).sort()) {
    sorted[key] = o[key];
}
return sorted;
```

File: lib/jsanalyze.js [hunk 2]
```diff
@@ -26,6 +26,23 @@ const path = require('path');
 const SOURCE_MAPPING_URL_REGEXP = /\/\/#[ \t]+sourceMappingURL=([^\s'"`]+?)[ \t]*$/mg;
 const __ = appc.i18n(__dirname).__;
 
+function sortObject (o) {
```
Comment (cb1kenobi) [inline lib/jsanalyze.js:29]: Please remove the space between the function and parens.

Comment (cb1kenobi) [inline lib/jsanalyze.js:29]: Export this function so we can use it in the test.

File: tests/jsanalyze_test.js [hunk 3]
```diff
@@ -6,6 +6,23 @@ const fs = require('fs-extra');
 const should = require('should'); // eslint-disable-line no-unused-vars
 const jsanalyze = require('../lib/jsanalyze');
 
+function sortObject (o) {
```
Comment (cb1kenobi) [inline tests/jsanalyze_test.js:9]: No need to duplicate code. Export `sortObject()` from `jsanalyze.js` and call `jsanalyze.sortObject()`.

File: tests/jsanalyze_test.js [hunk 4]
```diff
@@ -65,16 +82,14 @@ describe('jsanalyze', function () {
...
-		});
-
 		it('generates source maps inline into generated js file', function () {
 			const inputJSFile = path.join(__dirname, 'resources/input.js');
 			const contents = fs.readFileSync(inputJSFile, 'utf-8');
-			const expectedSourceMap = fs.readJSONSync(inputJSFile + '.map');
+			var expectedSourceMap = fs.readJSONSync(inputJSFile + '.map');
```
Comment (cb1kenobi) [inline tests/jsanalyze_test.js:88]: Try to get in the habit of using template literals:

```suggestion
			var expectedSourceMap = fs.readJSONSync(`${inputJSFile}.map`);
```

File: tests/jsanalyze_test.js [hunk 5]
```diff
@@ -88,9 +103,10 @@ describe('jsanalyze', function () {
 		it('generates source maps inline into generated js file and removes sourcesContent for android platform', function () {
 			const inputJSFile = path.join(__dirname, 'resources/input.js');
 			const contents = fs.readFileSync(inputJSFile, 'utf-8');
-			const expectedSourceMap = fs.readJSONSync(inputJSFile + '.map');
+			var expectedSourceMap = fs.readJSONSync(inputJSFile + '.map');
```
Comment (cb1kenobi) [inline tests/jsanalyze_test.js:106]: Use `let` instead of `var`. There's a couple occurrences below.

**Review by cb1kenobi — CHANGES_REQUESTED** [review]

File: lib/jsanalyze.js (hunk 1 above)
Comment (m1ga) [reply lib/jsanalyze.js]: :+1: fixed

File: lib/jsanalyze.js (hunk 2 above)
Comment (m1ga) [reply lib/jsanalyze.js]: :+1: fixed

File: tests/jsanalyze_test.js (hunk 3 above)
Comment (m1ga) [reply tests/jsanalyze_test.js]: :+1: fixed

File: tests/jsanalyze_test.js (hunk 4 above)
Comment (m1ga) [reply tests/jsanalyze_test.js]: :+1: fixed

File: tests/jsanalyze_test.js (hunk 5 above)
Comment (m1ga) [reply tests/jsanalyze_test.js]: :+1: fixed

**Comment by m1ga** [comment]
Thanks for the feedback :+1: Fixed all issues

**Review by cb1kenobi — APPROVED** [review]
Looks awesome! **APPROVED**!

**Pull request merged**